### DISCUSSION TITLE
NUCLEO-H755ZI

### DISCUSSION THREAD

**Discussion by lonejack** [cesanta/mongoose#2561] [Ask questions, get answers] [answered]
Hi,
I'm trying to implement the mongoose on STM32H755.
In all examples about STM32 it seems presents only single core app. Am I rigth?
If NO, can you suggest an example that exploit a dual core MCU?
The 755 has two cores (CM4,CM7). Based on the manual it seems more suitable to use the CM4 for the mongoose. Do you agree?
Thank you,
C.

**Comment by scaprile** [comment, marked as answer]
Mongoose is not an OS. Mongoose can run over an OS or can run in bare metal.
If you run on an OS/RTOS, it is this OS/RTOS that performs resource administration and you will configure your tasks as per that OS/RTOS API. Any related example will do.
If you do not run on an OS/RTOS, and you have chosen a multicore microcontroller, you should have a reason for doing that. Then, you should also have a reason to run the networking section in one core or the other because you have analyzed your resources and what you need to achieve and you know whether it is best for you to use one or the other for this or that. Without designing your application I can't say anything besides the obvious: at same clock freq, and with no accelerator, TLS runs faster on an M7.
As this is baremetal, and you will flash for one core or other core, you can choose the closest example. However, we have only tested our drivers and hardware initialization for those microcontrollers and boards we have examples for
- your microcontroller might have a different MAC controller, or your board a different PHY, or a different pinout, and we don't know, you do.
- your dual core might have some resource sharing/assignment to grab the MAC controller, or a different memory map, and we don't know, you do.
